### PR TITLE
Test additional Python versions

### PR DESCRIPTION
**Pull request by tonybaloney**
The documentation states that Python 3.8 is the tested version, but newer versions are recommended.

This PR adds the newer versions to the test matrix.

pyproject.toml has also been updated as it binds the python version to only the 3.8 series.

### CODE REVIEW

**Comment by squirrelsc** [comment]
They're waiting near one hour, is it caused by too many workflows in the pipeline?

**Comment by squirrelsc** [comment]
try to reopen the pr to trigger checks.

**Comment by tonybaloney** [comment]
Check that the "Approve and Run" button is not showing toward the bottom of this PR

**Review by squirrelsc — APPROVED** [review]

**Comment by squirrelsc** [comment]
> Check that the "Approve and Run" button is not showing toward the bottom of this PR

You are right, the button doesn't show on my page. Do you know how to fix it?

**Comment by tonybaloney** [comment]
@squirrelsc I think Github actions was broken. One way to fix this is to go and cancel the first actions run from https://github.com/microsoft/lisa/actions/runs/2579496054

**Comment by LiliDeng** [comment]
once it merges, does every job will be in queue like this?

**Comment by squirrelsc** [comment]
> once it merges, does every job will be in queue like this?

If we cannot figure out the problem, we shouldn't merge it.

**Pull request closed without merging**
